doc/api/readline.md

```diff
@@ -258,6 +258,36 @@ rl.question('What is your favorite food? ', (answer) => {
 });
 ```
 
+The `callback` function passed to `rl.question()` does not follow the typical
+pattern of accepting an `Error` object or `null` as the first argument.
+The `callback` is called with the provided answer as the only argument. If
+the question was canceled by `rl.questionCancel()` the callback will be called
+without arguments.
```

Comment:
Since the question is `canceled` IMO the callback shouldn't be called at all since it doesn't support error-conditions.
Or perhaps we could add a second argument to the callback though that is technically breaking.

Reply:
The use-case I was striving for is to easily wrap the question with a promise. It would of course be preferable to have had an error to emit to the callback but calling the callback without value is something that would happen only if `questionCancel` is called so, at least in my, view should not be a breaking change since it will not happen unless you already use the newer functionality of `questionCancel`.
But adding a second error callback would solve this just as well.
Maybe if we add support for a question[util.promisify.custom] it would make more sense, then the promisified version would reject/throw on cancel.
Not calling the callback, or an errorcallback, would require the caller of requestCancel to also notify the one asking the question in some other way, and in my view reducing the usefulness of this somewhat.

Reply:
> Not calling the callback, or an errorcallback, would require the caller of requestCancel to also notify the one asking the question in some other way, and in my view reducing the usefulness of this somewhat.

I considered this, but since the current callback is success-only and doesn't concern itself with any errors I wanted to leave it that way and just let the caller handle the case when they call `.questionCancel()`.

~~Regarding `would require the caller of requestCancel to also notify the one asking`, I didn't envision for this to be used cross-libraries so to speak (i.e. some library starts a question and the other somehow cancels it, is there such a use case?) and if we are the ones starting the question then we can handle the cancellation just fine.~~

~~I understand your concern but IMO empty callback may be more confusing for error handling, something like an event would be better I think.~~

I didn't consider the identification or mapping of a question to cancel, in the event case (or without an empty callback), we won't be able to know which question was canceled.
Yeah, I guess empty callback is the only option here (or adding another argument to the callback though that is both inconsistent and kind of breaking)

Reply:
One could add a completely new method also which follows the pattern and deprecate question. `ask` maybe :)
But if I understand you correctly you support the solution with calling the callback without arguments? (even though it is far from ideal)

Reply:
Yeah, though I guess we will be moving to streamline such cases with AbortController.